distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```diff
@@ -1,99 +1,343 @@
+
+       > Example
+         Generate a float between 2.3 and 3.4.
+
+              import Standard.Base.Random.Random
+
+              i = Random.float 2.3 3.4
+    float : Float -> Float -> Float
+    float (min:Float=0.0) (max:Float=1.0) = Random_Generator.global_random_generator.float min max
 
     ## GROUP Random
-       Gets the next random Float between 0 and 1.
-    float : Float
-    float self =
-        self.java_random.nextDouble
+       Return a random `Boolean` using the default instance of
+       `Random_Generator`.
+
+       > Example
+         Generate a random boolean.
+
+             import Standard.Base.Random.Random
+
+             b = Random.boolean
+    boolean : Boolean
+    boolean = Random_Generator.global_random_generator.boolean
+
+    ## GROUP Random
+       Return a random Gaussian-distributed `Float` with the specified mean and
+       standard deviation.
+
+       Arguments:
+       - standard_deviation: The standard deviation of the sampled distribution.
+       - mean: The mean of the sampled distribution.
+
+       > Example
+         Generate an gaussian float.
+
+              import Standard.Base.Random.Random
+
+              f = Random.gaussian
+    gaussian : Float -> Float -> Float
+    gaussian (standard_deviation:Float=1.0) (mean:Float=0.0) = Random_Generator.global_random_generator.gaussian standard_deviation mean
+
+    ## GROUP Random
+       Returns `count` indices sampled from the range 0 (inclusive) to n
+       (exclusive) without replacement.
+
+       Arguments:
+       - n: The end of the range to sample from (exclusive).
+       - count: The number of samples to take.
+
+       > Example
+         Generate 10 random indices between 0 and 100.
+
+              import Standard.Base.Random.Random
+
+              indices = Random.indices 100 10
+    indices : Integer -> Integer -> Vector Integer
+    indices (n:Integer) (count:Integer=1) = Random_Generator.global_random_generator.indices n count
 
     ## GROUP Random
-       Gets the next random Float from a normal distribution with mean 0 and std-dev 1.
-    gaussian : Float
-    gaussian self =
-        self.java_random.nextGaussian
+       Return a `Date` within the specified `Date_Range`.
+
+       Arguments:
+       - date_range: The `Date_Range` that specifies the range to pick from.
+
+       > Example
+         Generate a random date.
+
+             import Standard.Base.Random.Random
+
+             range = Date_Range.new (Date.new 2023 03 01) (Date.new 2023 10 15)
+             d = Random.date range
+
+       > Example
+         Generate a random date, with an inclusive end date.
+
+             import Standard.Base.Random.Random
+
+             range = my_first_day_of_vacation.up_to my_last_day_of_vacation include_end=True
+             d = Random.date range
+    date : Date_Range -> Date
+    date (date_range:Date_Range) = Random_Generator.global_random_generator.date date_range
```

Comment:
This feels mis-aligned with the others?

```
date (min:Date) (max:Date)
```
would seem more aligned.

Reply:
Done

Reply:
With this changed however, we again face the question, should `max` be exclusive?

I guess for consistency with other operations we need to keep it that way, but it feels a bit less intuitive IMO, as I was explaining in the 'find random date in range' examples above.

Reply:
Yes, for consistency. End-exclusive ranges are always a bit less intuitive for some users, but if we changed some we should change them all, not just for date/time ranges.

Reply:
We could add a parameter `end_exclusive`, possibly defaulting to `True`.

Reply:
Done